platforms/core-configuration/graph-serialization/src/main/kotlin/org/gradle/internal/serialize/graph/Contexts.kt

```diff
@@ -163,10 +174,40 @@ object InlineStringDecoder : StringDecoder {
 
     override fun readString(decoder: Decoder): String =
         decoder.readString()
+
+    override fun close() = Unit
```

Comment:
🤔 I'm not a big fan of optional `close` methods. I'd rather have the implementation classes themselves signal when they must be closed and be forced to implement `close`. It makes no sense for an `InlineStringEncoder` to be closed.

Reply:
I think the rationale here was to deal with the different kinds of special encoders/decoders in a more uniform way (without requiring to concrete types etc), but I may be misremembering